crates/bin/src/entry.rs

```diff
@@ -16,9 +17,61 @@ use log::{debug, error, info, warn, LevelFilter};
 use miette::{miette, Result, WrapErr};
 use tokio::task::block_in_place;
 
-use crate::{args::Args, install_path, ui::UIThread};
+use crate::{
+    args::{Args, Strategy},
+    install_path,
+    ui::UIThread,
+};
 
 pub async fn install_crates(mut args: Args, jobserver_client: LazyJobserverClient) -> Result<()> {
+    let strategies = args
+        .strategies
+        .take()
+        .map(|strategies| {
+            // Remove duplicate strategies
```

Comment:
We do have itertools in the dependencies already, so could use the utility from there, but I'm not fussed.

Reply:
I checked `IterTools` and I don't think it would help.

The code we have now is probably the most simple one and easy to maintain, while having good performance in practice.

Only thing we can improve is to replace `Vec` with `SmallVec`, which could save a heap allocation but I think that's just macro optimization.